yw_website/yw_website/apps/website/tests.py

```diff
@@ -44,50 +46,73 @@ def test_bad_yw_ping(self):
 
     def test_save_upload(self):
         route = '/save/'
+        now = datetime.datetime.now()
         data = {}
-        data['username'] = self.username
-        data['title'] = 'test_title'
-        data['description'] = 'test_description'
-        data['model'] = 'test_model'
-        data['model_checksum'] = '2341234123423'
-        data['graph'] = 'test_graph'
-        data['recon'] = 'test_recon'
-
-        response = self.client.post(route, data)
+        data["username"] = self.username
+        data["title"] = "test_title"
+        data["description"] = "test_description"
+        data["model"] = "test_model"
+        data["model_checksum"] = str(uuid.uuid1())
+        data["graph"] = "test_graph"
+        data["recon"] = "test_recon"
+        data["tags"] = ["tag_1", "tag_2", "tag_3"]
+        data["scripts"] = [{"name":"script_1", "checksum":str(uuid.uuid1()), "content":"script_1_content"},
+                            {"name":"script_2", "checksum":str(uuid.uuid1()), "content":"script_2_content"},
+                            {"name":"script_3", "checksum":str(uuid.uuid1()), "content":"script_3_content"}]
+        data["files"] = [{"name":"file_name_1", "checksum":str(uuid.uuid1()), "size":3, "uri":"file_uri1", "last_modified":now},
+                        {"name":"file_name_2", "checksum":str(uuid.uuid1()), "size":9, "uri":"file_uri2", "last_modified":now}]
+
+        response = self.client.post(route, data, format='json')
+        
         self.assertEquals(response.status_code, 200,
                           msg="Could not upload a workflow")
 
         data['username'] = 'nousername'
-        bad_response = self.client.post(route, data)
+        bad_response = self.client.post(route, data, format='json')
```

Comment:
it might be a good idea to separate a good and bad response in separate tests since we are testing different outcomes. useful for figuring out what part of your code is failing faster

Reply:
Agreed